src/imitation/scripts/common/reward.py

```diff
@@ -43,6 +53,8 @@ def config_hook(config, command_name, logger):
         res["net_cls"] = default_net
     if "normalize_input_layer" not in config["reward"]["net_kwargs"]:
         res["net_kwargs"] = {"normalize_input_layer": networks.RunningNorm}
+    if "normalize_output_layer" not in config["reward"]["net_kwargs"]:
+        res["net_kwargs"] = {"normalize_output_layer": None}
```

Comment:
`reward` ingredient is used in both `adversarial` and `PreferenceComparisons`. Enabling it by default might mean we need to change the code in `adversarial`. Do we want that in `GAIL` and `AIRL`?

Reply:
To change it to enabled by default, we might want to also delete the lines in [`imitation.algorithms.adversarials.common`](https://github.com/HumanCompatibleAI/imitation/blob/ed45793dfdd897d3ac1f3a863a8816b56d436887/src/imitation/algorithms/adversarial/common.py#L220).

For `AIRL`, it's fine to do this because the [`reward_train`](https://github.com/HumanCompatibleAI/imitation/blob/ed45793dfdd897d3ac1f3a863a8816b56d436887/src/imitation/algorithms/adversarial/airl.py#L90) of the `AdversarialTrainer` outputs the normalized output directly. Hence, it's equivalent.

For `GAIL`, the `AdversarialTrainer` would [first normalize the reward in `self.base.forward()`](https://github.com/HumanCompatibleAI/imitation/blob/ed45793dfdd897d3ac1f3a863a8816b56d436887/src/imitation/algorithms/adversarial/gail.py#L35), then take the negative log sigmoid of the logits. This order is different from the previous case of taking `-F.logsigmoid(logits)` first then normalize.

Reply:
I basically ignored it for now by checking asserting it's running `train_preference_comparisons`, and raise `NotImplementedError` when it's running `train_adversarial`.

Reply:
Oh good point. We can have different defaults in the config hook depending on algorithm. But it sounds like we do want it normalized even in AIRL/GAIL, just right now in GAIL the code is taking logits of whatever the reward network outputs, and we want normalization after that?

This is a bit tricky to do cleanly. We could make `LogSigmoidRewardNet` take `normalize_output_layer` as an option. But making the config handle this would still be awkward. We could have separate config options for GAIL normalization, but that's error prone. Alternative would probably mean branching on whether or not its GAIL, both in the sacred config and in `train_adversarial`.

Reply:
Sorry I was wrong about it. In GAIL, `train_gen()` and `train_disc()` used `reward_net.predict()` and `reward_net.forward()` respectively, and the normalization only applied to `reward_net.predict()`, so we are good.